### PR TITLE
project structure cleanup

### PR DESCRIPTION
**Pull request by ericbuckley**
## Description
Restructuring the codebase into `models/` and `schemas/`.  The former will be used for [sqlalchemy ORM models](https://www.sqlalchemy.org/) and the latter for [pydantic validation models](https://docs.pydantic.dev/latest/).  This is necessary so we create a strategy for NOT having circular imports.

## Related Issues
refs #65 

## Additional Notes
With the separation of models into 2 distinct packages, `recordlinker.models` and `recordlinker.schemas`, we will allow schemas to depend on models, but not the other way around.

- `BlockingKeys.to_value` was migrated to `PIIRecord.blocking_keys` to resolve the dependency issue
- There is still an issue with `Patient.record`, as this getter/setter has a dependency on the.  I'm not sure the best way to address this.  The code really isn't necessary, but its an optimization right now that is helpful.  I'm going to add an [issue to resolve this in the future](https://github.com/orgs/CDCgov/projects/34/views/1?sortedBy%5Bdirection%5D=asc&sortedBy%5BcolumnId%5D=Assignees&pane=issue&itemId=82498554).  Right now we can circumvent and potential problems by moving the import inside the function, but that is not ideal.

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.
- [x] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

**Review by cbrinson-rise8 — DISMISSED** [review]

File: src/recordlinker/models/algorithm.py [hunk 1]
```diff
@@ -12,9 +12,9 @@ class Algorithm(Base):
     id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
     is_default: orm.Mapped[bool] = orm.mapped_column(default=False, index=True)
     label: orm.Mapped[str] = orm.mapped_column(sqltypes.String(255), unique=True)
-    description: orm.Mapped[str] = orm.mapped_column(sqltypes.Text())
+    description: orm.Mapped[str] = orm.mapped_column(sqltypes.Text(), nullable=True)
```
Comment (cbrinson-rise8) [inline src/recordlinker/models/algorithm.py:15]: We might want to run an `alembic revision --autogenerate` since changes were made to the `algorithm` model

Comment (ericbuckley) [reply src/recordlinker/models/algorithm.py:15]: Yeah thats a fair point, I was lax on the alembic changes because of [this upcoming work](https://github.com/CDCgov/RecordLinker/issues/57).  I haven't been happy with it so far, and its honestly a bit early in the project for a tool like this, so I think we can remove for the time being.

**Review by cbrinson-rise8 — APPROVED** [review]
